### DISCUSSION TITLE
Error: Renderer process crashed (status 5)

### DISCUSSION THREAD

**Discussion by jasondh** [qutebrowser/qutebrowser#8226] [Q&A]
Just started getting this error on almost every web page I visit since upgrading Qutebrowser.
Anyone suggest a solution here please?

qutebrowser v3.2.0
Backend: QtWebEngine 6.7.1
Using MacOS 14.5 on M1 Mac.

**Comment by mschilli87** [comment]
Same here. MacOS 14.5 on M2 Mac. qutebrowser from homebrew.

**Comment by jasondh** [comment]
I submitted a report via the app.   Using Homebrew myself. 
Hopefully an update will be issued soon.

**Reply by The-Compiler** [reply]
This is an issue in the underlying QtWebEngine/Chromium and won't be something qutebrowser itself can fix. If I can reproduce it on my Mac I'll probably report it upstream, but other than that, little I can do most likely.

**Reply by toofar** [reply]
I feel like we could probably use a little more help figuring out what is going wrong! If I'm reading [the cask script](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/q/qutebrowser.rb) right it simply installs the .dmg from our releases page. That build isn't built for apple silicon, so I assume that means you need to run it via rosetta. But I tried to run the smoke test on the homebrew build on M1 without rosetta in github actions and I didn't see any segfaults (hard to see if it actually worked though), so maybe not?: https://github.com/toofar/qutebrowser/actions/runs/9433155394/job/25983768727

Anyway, things to try:
* run with `--qt-flag single-process`, and under gdb if you can. I think this is how we get stack traces for renderer crashes
* try installing the .dmg directly from our releases page instead of homebrew and see if that helps
* try installing Qt from homebrew but running qutebrowser from source (instead of using our release): https://qutebrowser.org/doc/install.html#_homebrew
* share any interesting logs and your full :version and if you have rosetta installed
* try the `qutebrowser-nightly-2024-06-08-3d6370b-macos` build from this job, it's built on macos-14, no idea if that'll help or not though https://github.com/qutebrowser/qutebrowser/actions/runs/9432304960

**Comment by mschilli87** [comment]
I can can cofirm that I had to install rosetta after updating to 3.2.0. Note that previous versions qutebrowser installed via homebrew did *not* require rosetta.
I'll can try some of the other things when I have time (will I need debg symbols for the stack trace?) but I am quite busy these days.

---
*edit*: My version info:


```
qutebrowser v3.2.0
Backend: QtWebEngine 6.7.1
  based on Chromium 118.0.5993.220
  with security patches up to 124.0.6367.202 (plus any distribution patches)
  (source: api)
Qt: 6.7.1

CPython: 3.12.1
PyQt: 6.7.0

Qt wrapper info:
  PyQt6: success
  PyQt5: not imported
  -> selected: PyQt6 (via autoselect)

colorama: 0.4.6
jinja2: 3.1.4
pygments: 2.18.0
yaml: 6.0.1
adblock: 0.6.0
objc: 10.3
PyQt6.QtWebEngineCore: 6.7.0
PyQt6.sip: 6.7.12
pdf.js: 4.3.136 (bundled)
sqlite: 3.45.3
QtNetwork SSL: OpenSSL 3.0.11 19 Sep 2023

Style: QMacStyle
Platform plugin: cocoa
OpenGL: Apple, 2.1 Metal - 88.1
Platform: macOS-14.5-x86_64-i386-64bit, 64bit
Frozen: True
Imported from /Applications/qutebrowser.app/Contents/Frameworks/qutebrowser
Using Python from /Applications/qutebrowser.app/Contents/MacOS/qutebrowser
Qt library executable path: /Applications/qutebrowser.app/Contents/Frameworks/PyQt6/Qt6/libexec, data path: /Applications/qutebrowser.app/Contents/Frameworks/PyQt6/Qt6
OS Version: 14.5, x86_64

Paths:
auto config: /Users/mschilli87/Library/Preferences/qutebrowser
cache: /Users/mschilli87/Library/Caches/qutebrowser
config: /Users/mschilli87/.qutebrowser
data: /Users/mschilli87/Library/Application Support/qutebrowser
runtime: /private/var/folders/vp/r4rpx_414pb5mg_3qsfm8l840000gp/T/qutebrowser

Autoconfig loaded: yes
Config.py: no config.py was loaded
Uptime: 21:16:09
```

**Reply by The-Compiler** [reply]
> I can can cofirm that I had to install rosetta after updating to 3.2.0. Note that previous versions qutebrowser installed via homebrew did _not_ require rosetta.

That makes no sense to me. There never was an Apple Silicon release of qutebrowser thus far (#6478), and in fact, Homebrew itself [added a caveat to tell users about this](https://github.com/Homebrew/homebrew-cask/pull/158242/files) back in October, for v3.0.2.

However, given I paid for access to a M1 machine to debug this, I decided to change that and add Apple Silicon builds for future releases.

All this doesn't explain this issue though (which I still can't reproduce). If you did indeed install previous versions via Homebrew (via Cask), I don't see a way how they could have worked without Rosetta.

**Reply by mschilli87** [reply]
Tbh I am not much of an Apple guy. 🤷 That machine is not mine, but I have to use it. I heard of Rosetta from the warning after `brew upgrade` concerning qutebrowser. Maybe I re-installted it thinking it was not installed before. I will do what I can to help debug this, but I don't expect anyone to spend their free time supporting propriatary systems, just to get that straight.

**Comment by mschilli87** [comment]
This is what I get from `pip install [--user] qutebrowser`:

```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.

    If you wish to install a Python library that isn't in Homebrew,
    use a virtual environment:

    python3 -m venv path/to/venv
    source path/to/venv/bin/activate
    python3 -m pip install xyz

    If you wish to install a Python application that isn't in Homebrew,
    it may be easiest to use 'pipx install xyz', which will manage a
    virtual environment for you. You can install pipx with

    brew install pipx

    You may restore the old behavior of pip by passing
    the '--break-system-packages' flag to pip, or by adding
    'break-system-packages = true' to your pip.conf file. The latter
    will permanently disable this error.

    If you disable this error, we STRONGLY recommend that you additionally
    pass the '--user' flag to pip, or set 'user = true' in your pip.conf
    file. Failure to do this can result in a broken Homebrew installation.

    Read more about this behavior here: <https://peps.python.org/pep-0668/>
```

Do you recommend I go ahead with `pip install --user --break-system-packages` anyhow?

**Reply by toofar** [reply]
No, you shouldn't need to do that! Either create a virtualenv and try to install and run qute while that is activated (like the example in the error message you posted), or install pipx from homebrew and use that to handle the virtualenv stuff in the background (oh, it also says that in the error message!).

**Reply by mschilli87** [reply]
Installing qutebrowser via `pipx qutebrowser` gives me `~/.local/bin/qutebrowser` but launching it gives me

```
Traceback (most recent call last):
  File "/Users/mschilli87/.local/bin/qutebrowser", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/Users/mschilli87/.local/pipx/venvs/qutebrowser/lib/python3.12/site-packages/qutebrowser/qutebrowser.py", line 228, in main
    earlyinit.early_init(args)
  File "/Users/mschilli87/.local/pipx/venvs/qutebrowser/lib/python3.12/site-packages/qutebrowser/misc/earlyinit.py", line 341, in early_init
    init_qtlog(args)
  File "/Users/mschilli87/.local/pipx/venvs/qutebrowser/lib/python3.12/site-packages/qutebrowser/misc/earlyinit.py", line 296, in init_qtlog
    from qutebrowser.utils import log, qtlog
  File "/Users/mschilli87/.local/pipx/venvs/qutebrowser/lib/python3.12/site-packages/qutebrowser/utils/qtlog.py", line 15, in <module>
    from qutebrowser.qt import core as qtcore
  File "/Users/mschilli87/.local/pipx/venvs/qutebrowser/lib/python3.12/site-packages/qutebrowser/qt/core.py", line 17, in <module>
    machinery.init_implicit()
  File "/Users/mschilli87/.local/pipx/venvs/qutebrowser/lib/python3.12/site-packages/qutebrowser/qt/machinery.py", line 278, in init_implicit
    raise NoWrapperAvailableError(info)
qutebrowser.qt.machinery.NoWrapperAvailableError: No Qt wrapper was importable.

Qt wrapper info:
  PyQt6: ModuleNotFoundError: No module named 'PyQt6'
  PyQt5: ModuleNotFoundError: No module named 'PyQt5'
  -> selected: None (via autoselect)
```

**Reply by toofar** [reply]
Hmmm, I guess either PyQt6 isn't installed or pipx isn't picking up the homebrew python path (which would have PyQT installed to it).
I don't have a setup to reproduce this but stack overflow says you can do this to add the homebrew installed python libraries to your path? (Assuming pipx and homebrew are using the same python version in the first place) https://stackoverflow.com/a/70683186

**Comment by mschilli87** [comment]
```sh
brew info pyqt
```

```
==> pyqt: stable 6.7.0 (bottled)
Python bindings for v6 of Qt
https://www.riverbankcomputing.com/software/pyqt/intro
Installed
/opt/homebrew/Cellar/pyqt/6.7.0 (1,229 files, 35.3MB) *
  Poured from bottle using the formulae.brew.sh API on 2024-06-09 at 09:21:56
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/p/pyqt.rb
License: GPL-3.0-only
==> Dependencies
Build: pyqt-builder ✘
Required: python@3.12 ✔, qt ✔
==> Analytics
install: 10,619 (30 days), 30,956 (90 days), 96,080 (365 days)
install-on-request: 5,515 (30 days), 14,869 (90 days), 46,724 (365 days)
build-error: 0 (30 days)
```

I installed `pipx from homebrew`, too, so I hope the python versions match...

**Reply by mschilli87** [reply]
Regarding your other suggestion, `which python` already points to `/opt/homebrew/opt/python@3.12/libexec/bin/python`, so modifying the `PATH` won't help.

**Reply by toofar** [reply]
I'm not sure where to go from here. You can go `python3 -c 'import PyQt6; print(PyQt6.__file__)'` to make sure it finds PyQt6 properly, but it probably will. Presumably pipx is being dumb. Doing `pipx uninstall qutebrowser && pipx install --system-site-packages qutebrowser` to make it link to the system site packages might help, but I have no idea if the homebrew stuff counts as system.
The other thing to try would be trying to run qutebrowser from a normal virtualenv, not pipenv.

**Reply by mschilli87** [reply]
`pipx uninstall qutebrowser && pipx install --system-site-packages qutebrowser` did the trick. Will test & report back.

**Reply by mschilli87** [reply]
Still getting the crashes with homebrew `[py]qt` and pipx qutebrowser.

**Reply by toofar** [reply]
Okay, thanks for checking! Seems indeed it might be some incompatibility with PyQt or webengine. You can try the [nightly](https://github.com/qutebrowser/qutebrowser/actions/runs/9432304960) that was build on macos 14 if you want, otherwise it seems we need to wrangle a stack trace from somewhere.

**Reply by mschilli87** [reply]
[qb.log](https://github.com/user-attachments/files/15751658/qb.log)
I downloaded the nightly but it won't start. I attached the stderr output.

**Reply by The-Compiler** [reply]
What do you mean by "won't start" precisely? The logs look like everything should be fine, and it also seems to run here from a quick test on a M1 Mac with macOS Sonoma.

**Reply by mschilli87** [reply]
I mean if I run `qutebrowser` in the terminal, the above output gets printed to stderr but no browser window is created. Eventually, the command exits and I am returned to the shell prompt. :shrug:

**Comment by mschilli87** [comment]
```sh
python3 -c 'import PyQt6; print(PyQt6.__file__)'
```

```
/opt/homebrew/lib/python3.12/site-packages/PyQt6/__init__.py
```

**Comment by The-Compiler** [comment]
I got access to a Mac running Sonoma, but unfortunately I can't reproduce:

![image](https://github.com/qutebrowser/qutebrowser/assets/625793/6d2b9cd7-4c5e-443c-b3b8-0fcd3abad8ba)

Based on a `pgrep oahd` it looks like Rosetta is installed on it, though...

**Reply by mschilli87** [reply]
This might be conincidence but I feel it crashes more often if I highlight and copy some text.

**Reply by The-Compiler** [reply]
It would be good to have something more detailed on how this can be reproduced. OP made it sound as if only loading any webpage would trigger the crash ("Just started getting this error on almost every web page I visit"), but apparently it's not that easy then?

On what websites does this happen exactly, and about how often does it take you to to trigger it?

**Reply by mschilli87** [reply]
It happpens on more or less every site. 1/5 navigation, when copying, when loading a video (youtube is unusable). I have to copy/paste into this very text box as there is a high chance it crashes before I finish. If I click on Preview, it dies 8/10 times. If you hit this, it is hard to miss. Just crashed again when I clicked on Reply to post this reply.

**Comment by jasondh** [comment]
[QtWebEngineProcess-2024-06-12-120646.txt](https://github.com/user-attachments/files/15804051/QtWebEngineProcess-2024-06-12-120646.txt)

My Mac Crash File details attached.

I can't seem to pin it down to when it happens.  Or any specific web site.  It might happen 4 or 5 times during a browsing session.

**Reply by The-Compiler** [reply]
Thanks! Unfortunately it doesn't look like that has a lot of details on why/how the crash happened either... :( And so far I still cannot reproduce this, even after trying on three different M1/M2 machines from two providers I was able to rent without paying for an entire month.

**Reply by mschilli87** [reply]
How are you installing Qt on those machines? Maybe it's a homebrew specific issue?

**Reply by The-Compiler** [reply]
If you `brew install qutebrowser`, that just uses the .dmg from the qutebrowser releases, which comes with its own Qt (it's a brew cask package).

On one machine I tried the .dmg manually (as I don't have admin rights), on the other two I installed Homebrew and did `brew install qutebrowser`.

**Comment by jasondh** [comment]
I am starting to think that this is an issue with another app I have installed - **Smart Scroll.**
https://www.marcmoini.com/sx_en.html
If I disable this app (and the way it controls the smooth scrolling on Qutebrowser), then I don't get the error.
I have emailed the Author to see If this is indeed the fault of Smart Scroll.

**Reply by mschilli87** [reply]
AFAIK I do not have this app installed...any way I can check?

**Reply by jasondh** [reply]
You would either have it or not I suppose. - Have you tried booting in Safe Mode - then trying to replicate this issue?
This would remove any user installed services/apps that may be set to auto-start or run in the background (like my Mouse control app).
